src/dcqc/updaters.py

```diff
@@ -0,0 +1,58 @@
+from collections import defaultdict
+from csv import DictWriter
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List
+
+from dcqc.parsers import CsvParser
+from dcqc.suites.suite_abc import SuiteABC
+
+
+@dataclass
+class CsvUpdater:
+    input_path: Path
+    output_path: Path
+    parser: CsvParser
+
+    def __init__(self, input_path: Path, output_path: Path):
+        self.output_path = output_path
+        self.input_path = input_path
+
+    def update(self, suites: List[SuiteABC]):
+        suite_dict = defaultdict(list)
+        # {url: [list_of_statuses]} data structure to allow for multi-file targets
+        # TODO add support for suites with multiple files in them (multi-file targets)
+        for suite in suites:
+            url = suite.target.files[0].url
+            status = suite.get_status()
+            suite_dict[url].append(status.value)
+        # Evaluate dcqc_status for each url
+        collapsed_dict = {}
+        for url, statuses in suite_dict.items():
+            if "RED" in statuses:
+                collapsed_dict[url] = "RED"
+            elif "AMBER" in statuses:
+                collapsed_dict[url] = "AMBER"
+            elif "GREEN" in statuses:
+                collapsed_dict[url] = "GREEN"
+            else:
+                collapsed_dict[url] = "NONE"
```

Comment:
Do we need to cover this test case? 

When will this be "NONE"?

Reply:
It would only be None in a case where all of the Suites for a file have not had any statuses assigned. I don't think that it is technically possible for this to be the case the way everything is currently written.

Reply:
What about for files that aren't supported in dcqc?

Reply:
unsupported file types just error out [here](https://github.com/Sage-Bionetworks-Workflows/py-dcqc/blob/c7262f647cf584123d5de5b4611d7bbd9127eeac/src/dcqc/file.py#L97)

Reply:
I am concluding that it should not be possible for the final status of a suite to be None, so I will remove this case handling.